log/DESIGN.md

```diff
@@ -0,0 +1,198 @@
+# Logs Bridge API
+
+OpenTelemetry Logs tracking issue at [#4696](https://github.com/open-telemetry/opentelemetry-go/issues/4696).
+
+## Abstract
+
+We propose adding a `go.opentelemetry.io/otel/log` Go module which will provide
+[Logs Bridge API](https://opentelemetry.io/docs/specs/otel/logs/bridge-api/).
+
+## Background
+
+They key challenge is to create a well-performant API compliant with the specification.
+Performance is seen as one of the most imporatant charactristics of logging libraries in Go.
+
+## Design
+
+This proposed design aims to:
+
+- be specification compliant,
+- have similar API to Trace and Metrics API,
+- take advantage of both OpenTelemetry and `slog` experience to achieve acceptable performance.
+
+### Module structure
+
+The Go module consits of the following packages:
+
+- `go.opentelemetry.io/otel/log`
+- `go.opentelemetry.io/otel/log/embedded`
+- `go.opentelemetry.io/otel/log/noop`
+
+### LoggerProvider
+
+The [`LoggerProvider` abstraction](https://opentelemetry.io/docs/specs/otel/logs/bridge-api/#loggerprovider)
+is defined as an interface.
+
+```go
+type LoggerProvider interface{
+	embedded.LoggerProvider
+    Logger(name string, options ...LoggerOption) Logger
+}
+```
+
+### Logger
+
+The [`Logger` abstraction](https://opentelemetry.io/docs/specs/otel/logs/bridge-api/#logger)
+is defined as an interface.
+
+```go
+type Logger interface{
+	embedded.Logger
+    Emit(ctx context.Context, options ...RecordOption)
+}
+```
+
+The `Logger` has `Emit(context.Context, options ...RecordOption` method.
+
+The options are used to set log record parameters e.g. `WithBody`, `WithTimestamp`.
+There would NOT be a `WithLogRecord` option.
```

Comment:
What's the purpose of exported `Record` and `NewRecord` if there is no `WithLogRecord`? Where is the `Record` used?

Can you show API usage examples? I can see the usage examples for the rejected approaches, but not for the proposed approach.

Reply:
I have the same question, could you please show the proposed approach instead of just the rejected one.

Reply:
I did my best to address it via 5456b9de460616f37fe8d06cacf5b555937eef55 4365860c10d4db73d4f6fed83585bb9389ca44e9

I think that the prototype is the next step to validate the design.